tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedAssertIndexExistsOperation.cs

```diff
@@ -0,0 +1,78 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedAssertIndexExistsOperation : IUnifiedSpecialTestOperation
+    {
+        private readonly string _collectionName;
+        private readonly string _databaseName;
+        private readonly string _indexName;
+
+        public UnifiedAssertIndexExistsOperation(
+            string databaseName,
+            string collectionName,
+            string indexName)
+        {
+            _databaseName = databaseName;
+            _collectionName = collectionName;
+            _indexName = indexName;
+        }
+
+        public void Execute()
+        {
+            var client = DriverTestConfiguration.Client;
+            var indexes = client.GetDatabase(_databaseName).GetCollection<BsonDocument>(_collectionName).Indexes.List().ToList();
```

Comment:
Line 42 feels like it is doing way too much work and each step along the way is at a very different conceptual level. Consider adding a few intermediate variables:

    var client = DriverTestConfiguration.Client;
    var database = client.GetDatabase(_databaseName);
    var collection = database.GetCollection<BsonDocument>(_collectionName);
    var indexes = collection.Indexes.List().ToList();
    var indexNames = indexes.Select(i => i["name"].AsString);

Reply:
Or at the very least consider:

    var client = DriverTestConfiguration.Client;
    var collection = client.GetDatabase(_databaseName).GetCollection<BsonDocument>(_collectionName);
    var indexes = collection.Indexes.List().ToList();
    var indexNames = indexes.Select(i => i["name"].AsString);

so that we separate getting the `collection` from **executing** an operation on the collection.